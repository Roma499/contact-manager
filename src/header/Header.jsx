import React from 'react';
import { Link } from 'react-router-dom';
import Grid from '@material-ui/core/Grid';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import './Header.css';

function Header() {
  return (
    <AppBar position="static">
      <Toolbar>
        <Grid container spacing={16} alignItems="center" direction="row" justify="space-between">
          <Link to="/" className="app-header-link">
            <Typography variant="title" color="inherit">
              Contacts
            </Typography>
          </Link>
          <Link to="/contacts/new" className="app-header-link">
            + Add Contact
          </Link>
        </Grid>
      </Toolbar>
    </AppBar>
  );
}

export default Header;
