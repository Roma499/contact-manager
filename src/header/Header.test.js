import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Header from './Header';

it('renders without crashing', () => {
  shallow(<Header />);
});

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <Header />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
