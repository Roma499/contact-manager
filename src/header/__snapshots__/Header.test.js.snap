// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorPrimary-7"
>
  <div
    className="MuiToolbar-root-36 MuiToolbar-regular-38 MuiToolbar-gutters-37"
  >
    <div
      className="MuiGrid-container-40 MuiGrid-spacing-xs-16-63 MuiGrid-align-items-xs-center-48 MuiGrid-justify-xs-space-between-59"
    >
      <a
        className="app-header-link"
        href="/"
        onClick={[Function]}
      >
        <h2
          className="MuiTypography-root-137 MuiTypography-title-143 MuiTypography-colorInherit-156"
        >
          Contacts
        </h2>
      </a>
      <a
        className="app-header-link"
        href="/contacts/new"
        onClick={[Function]}
      >
        + Add Contact
      </a>
    </div>
  </div>
</header>
`;
